jsx/Form.js

```diff
@@ -1009,61 +1020,72 @@ class DateElement extends Component {
     }
   }
 
+  handleButton(e) {
+    let date = new Date();
+    let dateString = new Date(date.getTime() - (date.getTimezoneOffset() * 60000 ))
+                        .toISOString()
+                        .split('T')[0];
+    this.props.onUserInput(this.props.name, dateString);
+  }
+
   handleChange(e) {
-    this.props.onUserInput(this.props.name, e.target.value, e.target.id, 'date');
+    this.props.onUserInput(this.props.name, e.target.value);
   }
 
   render() {
     let disabled = this.props.disabled ? 'disabled' : null;
     let required = this.props.required ? 'required' : null;
     let requiredHTML = null;
+    let errorMessage = null;
+    let elementClass = null;
 
     // Add required asterix
     if (required) {
       requiredHTML = <span className="text-danger">*</span>;
     }
 
-    // Check if props minYear and maxYear are valid values if supplied
-    let minYear = this.props.minYear;
-    let maxYear = this.props.maxYear;
-    if (this.props.minYear === '' || this.props.minYear === null) {
-      minYear = '1000';
-    }
-    if (this.props.maxYear === '' || this.props.maxYear === null) {
-      maxYear = '9999';
+    // Add error message
+    if (this.props.errorMessage) {
+      errorMessage = <span>{this.props.errorMessage}</span>;
+      elementClass = 'has-error';
     }
 
-    // Handle date format
-    let format = this.props.dateFormat;
-    let inputType = 'date';
-    let minFullDate = minYear + '-01-01';
-    let maxFullDate = maxYear + '-12-31';
-    if (!format.match(/d/i)) {
-      inputType = 'month';
-      minFullDate = minYear + '-01';
-      maxFullDate = maxYear + '-12';
-    }
+    const todayButton = this.props.today && (
+      <Button
+        label='Today'
+        type="button"
+        onClick={this.handleButton}
+      />
+    );
```

Comment:
not in **24.1**

_added functionality to apply current date by clicking button_

to include as **CBIG override** with plans to push to **main** in some capacity